internal_worker.go

```diff
@@ -925,50 +1029,41 @@ func newAggregatedWorker(
 
 	processTestTags(&wOptions, &workerParams)
 
-	env := getHostEnvironment()
+	hostEnv := getHostEnvironment()
 	// workflow factory.
 	var workflowWorker Worker
 	if !wOptions.DisableWorkflowWorker {
-		if env.lenWorkflowFns() > 0 {
-			workflowFactory := newRegisteredWorkflowFactory()
-			testTags := getTestTags(wOptions.BackgroundActivityContext)
-			if testTags != nil && len(testTags) > 0 {
-				workflowWorker = newWorkflowWorkerWithPressurePoints(
-					workflowFactory,
-					service,
-					domain,
-					workerParams,
-					testTags,
-				)
-			} else {
-				workflowWorker = newWorkflowWorker(
-					getWorkflowDefinitionFactory(workflowFactory),
-					service,
-					domain,
-					workerParams,
-					nil)
-			}
+		testTags := getTestTags(wOptions.BackgroundActivityContext)
+		if testTags != nil && len(testTags) > 0 {
+			workflowWorker = newWorkflowWorkerWithPressurePoints(
+				service,
+				domain,
+				workerParams,
+				testTags,
+				hostEnv,
+			)
 		} else {
-			logger.Warn("Workflow worker is enabled but no workflow is registered. Use cadence.RegisterWorkflow() to register your workflow.")
```

Comment:
It might be a false warning here depending on how their code is set up. For instance they might not have Register in init function. What we need to do is add this warning in Start. If the worker is started and no workflows/activities are registered, issue a warning. I can add that.